README.md

```diff
@@ -28,13 +27,298 @@ pip install -U pyinstaller-hooks-contrib
+or a wrapper library for some web API may require credentials (and possibly a paid subscription) to test.
+In such cases, don't provide a test.
+Instead explain in the commit message why an automatic test is impractical then skip on to [the next step](#run-linter).
+
+
+#### Write tests(s)
+
+A test should be the least amount of code required to cause a breakage
+if you do not have the hook which you are contributing.
+For example if you are writing a hook for a library called `foo`
+which crashes immediately under PyInstaller on `import foo` then `import foo` is your test.
+If `import foo` works even without the hook then you will have to get a bit more creative.
+Good sources of such minimal tests are the hello world examples
+from the documentation of whichever library you're writing a hook for.
+Tests should not explicitly check for presence of data files or hidden modules directly -
+rather they should use parts of the library which are expected to use said data files or hidden modules.
+
+Tests currently all live in [src/_pyinstaller_hooks_contrib/tests/test_libraries.py](../master/src/_pyinstaller_hooks_contrib/tests/test_libraries.py).
+Navigate there and add something like the following, replacing all occurrences of `foo` with the real name of the library.
+(Note where you put it in that file doesn't matter.)
+
+```python
+@importorskip('foo')
+def test_foo(pyi_builder):
+    pyi_builder.test_source("""
+
+	# Your test here!
+        import foo
+
+	foo.something_fooey()
+
+    """)
+```
+
+If need be, you can specify version constraints using `@importorskip('foo', minversion="1.2.3")`
+if the library has changed significantly between versions.
+
+
+#### Run the test locally
+
+Running our full test suite is not recommended as it will spend a very long time testing code which you have not touched.
+Instead, run tests individually using either the `-k` option to search for test names:
+
+```
+pytest -k test_foo
+```
+
+Or using full paths:
+```
+pytest src/_pyinstaller_hooks_contrib/tests/test_libraries.py::test_foo
+```
+
+
+#### Run the test on CI/CD
+
+To test hooks on all platforms we use Github's continuous integration (CI/CD).
+Our CI/CD is a bit unusual in that it's triggered manually and takes arguments
+which limit which tests are run.
+This is for the same reason we filter tests when running locally -
+the full test suite runs like a dead snail on ketamine.
+
+First push the changes you've made so far.
+
+```commandline
+git push --set-upstream origin hook-for-foo
+```
+
+Replace *billy-the-buffalo* with your Github username in the following url then open it.
+It should take you to the `oneshot-test` actions workflow on your fork.
+You may be asked if you want to enable actions on your fork - say yes.
+```
+https://github.com/billy-the-buffalo/pyinstaller-hooks-contrib/actions/workflows/oneshot-test.yml
+```
+
+Find the **Run workflow** button and click on it.
```

Comment:
You need to click on 'Oneshot test' on the left to have the run button appear. I'd clarify not to click the 'New Workflow' button since that was my best guess at the next step after I couldn't find the run button.

Reply:
The url (having the `workflows/oneshot-test.yml` suffix) should've taken care of that. Did you have to click the *enable actions on this fork* page and that didn't forward you directly to the workflow?

Reply:
Yes, that's why

Reply:
Reckon 7f092f00acf50039f6b199a65103a142533658bc would have kept you right the first time?